src/hooks/useWalletSupportsChain/useWalletSupportsChain.ts

```diff
@@ -29,27 +29,41 @@ import {
   supportsPolygon,
   supportsThorchain,
 } from '@shapeshiftoss/hdwallet-core'
+import { isLedger } from '@shapeshiftoss/hdwallet-ledger'
 import { MetaMaskShapeShiftMultiChainHDWallet } from '@shapeshiftoss/hdwallet-shapeshift-multichain'
 import { useMemo } from 'react'
 import { useIsSnapInstalled } from 'hooks/useIsSnapInstalled/useIsSnapInstalled'
+import { selectAccountIdsByChainId } from 'state/slices/portfolioSlice/selectors'
+import { useAppSelector } from 'state/store'
 
 type WalletSupportsChainArgs = {
   isSnapInstalled: boolean | null
+  chainAccountIds: AccountId[] // allows dynamic chain-support detection for Ledger
   chainId: ChainId
   wallet: HDWallet | null
 }
 
 // use outside react
 export const walletSupportsChain = ({
   chainId,
+  chainAccountIds,
   wallet,
   isSnapInstalled,
 }: WalletSupportsChainArgs): boolean | null => {
   if (!wallet) return false
   const isMetaMaskMultichainWallet = wallet instanceof MetaMaskShapeShiftMultiChainHDWallet
   // Naming is slightly weird there, but the intent is if this evaluates to false, it acts as a short circuit
-  const shortCircuitFeatureDetection =
-    !isMetaMaskMultichainWallet || (isMetaMaskMultichainWallet && isSnapInstalled)
+  const shortCircuitFeatureDetection = (() => {
+    if (Boolean(isLedger(wallet) && !chainAccountIds.length)) return false
+    if (isMetaMaskMultichainWallet && !isSnapInstalled) return false
+    if (!isMetaMaskMultichainWallet) return true
+
+    // Do *not* short circuit the supportsXYZ methods
+    // This is either a Ledger with supported chain account ids, a MM wallet with snaps installed, or
+    // any other wallet, which supports static wallet feature-detection
+    return true
+  })()
+
```

Comment:
The guts of this PR - added Ledger checks here, and made the short-circuit logic cleaner and more commented to better convey the intent